### ISSUE TITLE
Add telemetry probe for startup time

### ISSUE THREAD

**Issue by mcomella** [mozilla-mobile/fenix#10069]
We should ensure this probe matches how we measure startup time in CI testing, if possible.

- - -
We repurposed https://github.com/mozilla-mobile/fenix/issues/8803 so this probe is to revisit the telemetry probe for startup. Now that we better know how to write telemetry, we probably want to do the minimal thing here rather than the large number of steps mentioned below.

### Why/User Benefit/User Problem
Startup time is an oft cited reason for friction and churn. We need to understand how long it takes Fenix to startup for our users so that we can prioritize work to address regressions and outliers that result in decreased engagement or churn. Local and CI-based testing provides an incomplete picture of how Fenix performs at startup.

### Impact
Improved understanding of how Fenix performs on startup for our users.

### Acceptance Criteria (how do I know when I’m done?)
We should make a best-effort to _incrementally_ instrument as much of startup as possible - until we hit diminishing returns - because it's impossible to do it all. Unfortunately, it will be difficult to define what this looks like in advance so take this as a guiding principle. Msg mcomella if this does not make sense.

Some incremental steps we should try to implement for GA:

* ~~Add a metric that tracks the start time of GV~~ I believe this is already available in Gecko telemetry
* Add a metric that tracks the duration of `FenixApplication.onCreate` (this may or may not be helpful but we want to add it to see if it is in practice)
* Add a metric that tracks `FenixApplication.onCreate` until first frame drawn
* Distinguish between first run and other cold starts (don't let this impact performance!)
* (is this reasonable?) Distinguish between which page was shown, i.e. which startup path was used
* Time from first frame drawn to reportFullyDrawn? Essentially, capture loading top sites & open tabs because this is dependent on content quantity

Some incremental steps we probably can't implement by GA but we should verify:

* Add a probe that tracks `FenixApplication.onCreate` until visual completeness
* Add warm/hot start times, distinguish from cold start
* Distinguish between cold starts after upgrading and other cold starts

Here is ecsmyth's ideal startup time probe that inspired these incremental steps:
> Glean telemetry probe landed in Fenix release that measures time, as closely as is practical, from when user initiates startup until the app is started and the initial screen is visually complete. The probe should differentiate between hot, cold, and warm startup scenarios and account for differences in the first page rendered (e.g., how desktop measures first paint and first paint of about:home)

**Comment by sraturi** [comment]
TLDR from a conversation with Glean:

It is okay for now to report the launch time as an event extra key (type safety from Fenix side). There is some work done on glean to have type safety for extra_keys. Once that is finished, we can implement type safety for this metric (given we don't change metric or keys name) without disrupting data flow.
There is also a struct metric type (not implemented yet) that could be a good alternative for this. However, there will be data disruption if we change the metric type in the future

**Comment by mcomella** [comment]
Merged with https://github.com/mozilla-mobile/fenix/pull/15605

**Comment by mcomella** [comment]
QA: we have a telemetry metric called `events.app_opened_all_startup`. It generally gets queued when the app is closed but will sometimes occur just after. In this issue, we added an extra, `first_frame_pre_draw_nanos`, which is roughly the time, in nanoseconds, from the app opening until the first frame is drawn. Please validate:
- the numbers seem reasonable based on the start-up type (i.e. the values in the other extras)
- the documentation for this extra makes sense & is usable by data scientists, engineers, product managers, etc.
